server/inventory_item/updater.py

```diff
@@ -0,0 +1,88 @@
+from apscheduler.schedulers.background import BackgroundScheduler
+from apscheduler.jobstores.mongodb import MongoDBJobStore
+from apscheduler.events import EVENT_JOB_EXECUTED, EVENT_JOB_ERROR
+import pymongo as pym
+from django_server.load_csv_to_db import main
+
+
+class Singleton(type):
+    """
+    A metaclass that ensures the singleton restriction
+    """
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class Scheduler(metaclass=Singleton):
+    """
+    Creates an object that holds the scheduler
+    """
+
+    def __init__(self, db_host, db_name):
+        # Making connection to mongoclient
+        client = pym.MongoClient(db_host)
+
+        job_stores = {
+            'default': MongoDBJobStore(database=db_name,
+                                       collection="jobs", client=client)
+        }
+
+        # This will get you a BackgroundScheduler with a MongoDBJobStore
+        # named “default” to replace the default one  and a ThreadPoolExecutor
+        # named “default” with a default maximum thread count of 10.
+        self.scheduler = BackgroundScheduler(jobstores=job_stores)
+
+
+scheduler = Scheduler("mongodb://localhost:27017/", "alta_development").scheduler
```

Comment:
why aren't you including this in the singleton?

Reply:
I can do it, it's just that if we need to update the db name or db url i thought it would just be easier to pass them as arguments instead of going manually change them in the singleton. What do you think?

Reply:
I don't see how having them here changes how you define it at all.

Just define it as a variable in the above class. Or am I missing something?

Reply:
Okay